From 59ef1ae1c2b97e2f50663e5c5a7f42a0b714c5e4 Mon Sep 17 00:00:00 2001
From: Tyler Gunn <tgunn@google.com>
Date: Mon, 7 Jun 2021 10:37:20 -0700
Subject: [PATCH] Add intent-filter priority for CONFIGURE_VOICEMAIL intent.

The android.telephony.action.CONFIGURE_VOICEMAIL intent is only handled
by the platform, so setting an intent filter priority here will ensure
that no other app can claim to handle it.

Test: Used test app which attempts to handle the intent to verify no
intent disambiguation dialog is shown and platform handles.
Bug: 185126149
Merged-In: I1c1b800737db5397d3e600e30c12ed2c8fc8efd9
Change-Id: I1c1b800737db5397d3e600e30c12ed2c8fc8efd9
(cherry picked from commit 5cd723c11bfe29dd44e7fcdc730c945d7348f3b6)
(cherry picked from commit 5521d7fce31260abde0b877ef9972658d87963fe)
---
 AndroidManifest.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 340aab704..e99561fdc 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -585,7 +585,7 @@
                 <action android:name="com.android.phone.CallFeaturesSetting.ADD_VOICEMAIL" />
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
-            <intent-filter>
+            <intent-filter android:priority="1000">
                 <action android:name="android.telephony.action.CONFIGURE_VOICEMAIL" />
                 <category android:name="android.intent.category.DEFAULT" />
             </intent-filter>
--
2.17.1
